pkg/controller/storagecluster/storagecluster.go

```diff
@@ -221,6 +223,53 @@ func (c *Controller) validate(cluster *corev1.StorageCluster) error {
 	if err := c.validateSingleCluster(cluster); err != nil {
 		return err
 	}
+	if err := c.validateTLSSpecs(cluster); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) validateTLSSpecs(current *corev1.StorageCluster) error {
+	if pxutil.IsTLSEnabledOnCluster(&current.Spec) {
+		tls := current.Spec.Security.TLS
+		// validate that any file paths are rooted to our mounted folder: /etc/pwx
+		if err := validateCertLocation("rootCA", tls.RootCA); err != nil {
+			return err
+		}
+		if err := validateCertLocation("serverCert", tls.ServerCert); err != nil {
+			return err
+		}
+		if err := validateCertLocation("serverKey", tls.ServerKey); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateCertLocation(certName string, certLocation *corev1.CertLocation) error {
+	// pxutil.IsEmptyOrNilCertLocation will also catch this scenario where no file specified, and only one of server cert or key specified
+	//   special casing here because we want a more helpful message than "must specify location of tls certificate as either file or kubernetes secret"
+	if certLocation != nil && pxutil.IsEmptyOrNilStringPtr(certLocation.FileName) && certLocation.SecretRef != nil && pxutil.IsPartialSecretRef(certLocation.SecretRef) {
+		return fmt.Errorf("%s: for kubernetes secret must specify both name and key", certName)
+	}
+
+	// should not be nil
+	if pxutil.IsEmptyOrNilCertLocation(certLocation) {
+		return fmt.Errorf("%s: must specify location of tls certificate as either file or kubernetes secret", certName)
+	}
+
+	// should not specify both file location and k8s secret
+	if !pxutil.IsEmptyOrNilSecretReference(certLocation.SecretRef) && !pxutil.IsEmptyOrNilStringPtr(certLocation.FileName) {
+		return fmt.Errorf("%s: can not specify both filename and secretRef as source for tls certs. ", certName)
+	}
+	// file must be under mounted folder /etc/pwx
+	if !pxutil.IsEmptyOrNilStringPtr(certLocation.FileName) {
+		if strings.HasPrefix(*certLocation.FileName, supportedRootFolderForTLSCerts) {
+			return nil // all good
+		}
+		return fmt.Errorf("%s: file path (%s) not under folder %s", certName, *certLocation.FileName, supportedRootFolderForTLSCerts)
+
+	}
```

Comment:
Let's not add this check here as it is specific to Portworx. We should create a TLS component under portworx driver and add this validation there.